tensordict/tensorclass.py

```diff
@@ -3296,6 +3296,14 @@ def __call__(self, *args, **kwargs):
         """Calling a NonTensorDataBase falls back to a call of its data."""
         return self.data(*args, **kwargs)
 
+    def __getitem__(self, idx):
+        if isinstance(self.data, list):
+            new_data = [self.data[i] for i in torch.as_tensor(idx).tolist()]
```

Comment:
Unfortunately, no, a list should not be treated differently than any other non-tensor data. 
For example, what happens if you set a list that has only one element in a tensordict of batch-size 3? 
Why list and not tuples? Or other indexable sequences?

`NonTensorStack` is the way to implement this behaviour.

You can have a non tensor data item shared among many different elements of a stack, this is why you can set a batch size to `NonTensorData` with no requirement on the data characteristics.